### PR TITLE
bugfix #2914 – suspense and router improvements

### PR DESCRIPTION
**Pull request by winniepukki**
Fixes scandipwa/scandipwa#2914

### CODE REVIEW

File: packages/scandipwa/src/util/Product/Product.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ export const getIndexedAttributeOption = (option) => {
 };
 
 /** @namespace Util/Product/getIndexedAttributes */
-export const getIndexedAttributes = (attributes) => attributes.reduce((indexedAttributes, attribute) => {
+export const getIndexedAttributes = (attributes = []) => attributes?.reduce((indexedAttributes, attribute) => {
```
Comment (zans-laksa) [inline packages/scandipwa/src/util/Product/Product.js:63]: This is not related to the task.

Comment (winniepukki) [reply packages/scandipwa/src/util/Product/Product.js]: Done ✅

**Comment by winniepukki** [comment]
@zans-laksa ping-ping

**Review by zans-laksa — APPROVED** [review]
Code review passed.
